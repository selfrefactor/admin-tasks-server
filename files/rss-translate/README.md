```
@Post('rss-translate')
  async rssTranslateRoute(@Body() input: {id: number}, @Res() res: Response) {
    return res.status(200).send('rss')
    if (!input) return res.status(400).send()
    const bookIndex = defaultTo(0, Number(input.id))
    const result = await this.speedReader.readBook(bookIndex)
    if (!result) return res.status(400).send()

    return res.status(200).send(result)
  }
```


nest-cli.json

```
 "rss-translate": {
      "type": "library",
      "root": "libs/rss-translate",
      "entryFile": "index",
      "sourceRoot": "libs/rss-translate/src",
      "compilerOptions": {
        "tsConfigPath": "libs/rss-translate/tsconfig.lib.json"
      }
    }
```
