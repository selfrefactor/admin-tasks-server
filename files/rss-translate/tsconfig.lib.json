{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "outDir": "../../dist/libs/rss-translate"
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules", "dist", "test", "**/*spec.ts"]
}
