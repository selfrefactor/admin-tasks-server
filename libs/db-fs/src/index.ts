export * from './db-fs.module';
export * from './db-fs.service';
