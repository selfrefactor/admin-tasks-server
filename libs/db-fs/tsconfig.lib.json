{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "outDir": "../../dist/libs/db-fs"
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules", "dist", "test", "**/*spec.ts"]
}
