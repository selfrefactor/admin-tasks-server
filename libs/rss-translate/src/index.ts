export * from './rss-translate.module';
export * from './rss-translate.service';
