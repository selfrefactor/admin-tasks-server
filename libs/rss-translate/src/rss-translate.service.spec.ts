import { Test, TestingModule } from '@nestjs/testing';
import { RssTranslateService } from './rss-translate.service';
import { DbFsService } from '../../db-fs/src/db-fs.service';

describe('RssTranslateService', () => {
  let service: RssTranslateService;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [RssTranslateService, DbFsService],
    }).compile();

    service = module.get<RssTranslateService>(RssTranslateService);
  });

  it('should be defined', () => {
    expect(service).toBeDefined()
  })

  it('read 1', async () => {
    const result = await service.read(`https://www.zdnet.de/feed/`, 'zdnet')
    // expect(
    //   result
    // ).toMatchSnapshot()
  });
  it.skip('read', async () => {
    const result = await service.read(`https://t3n.de/tag/rss/rss.xml`,'threen')
    // expect(
    //   result
    // ).toMatchSnapshot()
  });
})
;
