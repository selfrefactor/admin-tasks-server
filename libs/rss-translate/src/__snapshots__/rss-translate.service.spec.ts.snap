// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RssTranslateService read 1`] = `
Array [
  "https://t3n.de/news/dezentralisierung-chrome-rss-gut-1380199/",
  "https://t3n.de/news/slack-messenger-25-tipps-tricks-661440/",
  "https://t3n.de/news/erweiterungen-firefox-1165556/",
  "https://t3n.de/news/tipps-content-diebstahl-1025210/",
  "https://t3n.de/news/rss-reader-feedly-340092/",
  "https://t3n.de/news/10-nuetzliche-slack-erweiterungen-587167/",
  "https://t3n.de/news/serie-arbeit-motivation-5-753879/",
  "https://t3n.de/news/facebook-instant-articles-anlegen-696589/",
  "https://t3n.de/news/reclaim-data-tools-daten-kontrolle-datenhoheit-555073/",
  "https://t3n.de/magazin/tipps-redaktionsplan-content-marketing-233327/",
  "https://t3n.de/news/hootsuite-beginners-guide-479218/",
  "https://t3n.de/news/programmierst-rss-reader-30-464628/",
  "https://t3n.de/news/weg-google-co-hostest-alle-451096/",
  "https://t3n.de/magazin/seo-besseres-google-ranking-wordpress-230243/",
  "https://t3n.de/magazin/news-aggregatoren-vergleich-web-magazin-231139/",
  "https://t3n.de/magazin/typo3-entwicklung-typo3-sozialen-netzwerken-verknupfen-229577/",
  "https://t3n.de/news/pinterest-tracking-findest-375393/",
  "https://t3n.de/news/google-rss-einfach-erzeugst-367675/",
  "https://t3n.de/news/kundenbeziehungen-e-mail-newsletter-intensivieren-315016/",
  "https://t3n.de/magazin/twitter-facebook-joomla-anbinden-joomla-social-web-224713/",
]
`;
